packages/moonstone/ContextualPopupDecorator/ContextualPopupDecorator.js

```diff
@@ -495,6 +482,23 @@ const Decorator = hoc(defaultConfig, (config, Wrapped) => {
 			forward('onClose')
 		)
 
+		handleOpen = () => {
+			forward('onOpen', this.props);
+			this.positionContextualPopup();
+			const current = Spotlight.getCurrent();
+			this.updateLeaveFor(current);
+			this.setState(() => ({
```

Comment:
This (and on line 497) may be better with passing the object rather than a function since the new state isn't dependent upon the previous. WDYT?

Reply:
No objection here. (Though I am curious though how react handles it differently)

Reply:
Probably no effective difference beyond ms or less perf improvement avoiding creating the function and calling it. For me, I think the readability benefit is more compelling.